### PR TITLE
Divide by zero in Exif

### PR DESCRIPTION
**Pull request by wiredfool**
Not quite complete version of #1492 which preserves the semantics of 0/0 when that's what's actually in the file. It passes the existing test suite, with the exception of python 2.6.

* [x] Needs specific tests for the bug being fixed
* [x] Needs more detailed tests to define the behavior of the class that I've added.
* [x] Needs to work on 2.6
 
Please try this PR and let me know if it works for your use case.

### CODE REVIEW

**Comment by diwu1989** [comment]
I'm hitting a situation where I only want to read the rotation data from the exif but this divide by zero from the other fields is causing me to crash

Definitely want to see this merged in, just so that we can at least still read the other exif fields correctly

File: PIL/TiffImagePlugin.py [hunk 1]
```diff
@@ -225,6 +225,75 @@ def _limit_rational(val, max_val):
...
+
+        if denominator == 0:
+            self._val = float('nan')
+            return
+
+        try:
+            if denominator == 1:
+                self._val = Fraction(value)
+            else:
+                self._val = Fraction(value, denominator)
```
Comment (diwu1989) [inline PIL/TiffImagePlugin.py]: This is the test failure from python 2.6
 ```
======================================================================
ERROR: TestFileTiff.test__limit_rational_float
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/python-pillow/Pillow/Tests/test_file_tiff.py", line 310, in test__limit_rational_float
    ret = _limit_rational(value, 65536)
  File "/home/travis/build/python-pillow/Pillow/PIL/TiffImagePlugin.py", line 219, in _limit_rational
    n_d = IFDRational(1 / val if inv else val).limit_rational(max_val)
  File "/opt/python/2.6.9/lib/python2.6/fractions.py", line 99, in __new__
    numerator = operator.index(numerator)
TypeError: 'float' object cannot be interpreted as an index
```
so the numerator in this situation `value` should be cast into an int please

File: PIL/TiffImagePlugin.py [hunk 2]
```diff
@@ -225,6 +225,75 @@ def _limit_rational(val, max_val):
...
+        except:
+            print(type(value), type(denominator))
```
Comment (diwu1989) [inline PIL/TiffImagePlugin.py]: is it common for Pillow code to have print statements?

Comment (wiredfool) [inline PIL/TiffImagePlugin.py]: It is when I need to debug something.

File: PIL/TiffImagePlugin.py (hunk 1 above)
Comment (wiredfool) [inline PIL/TiffImagePlugin.py]: The problem with Py2.6 is that `Fraction` got a lot better in 2.7, and will take almost any number in the first position. In 2.6, you have to use the `Fraction.from_float` method (or similar)

Casting to an `int` there is going to return incorrect values.

**Comment by willmcgugan** [comment]
Using Fraction or another object for rationals makes the data returned by _getexif not trivially serializable by JSON for example.

Any chance you could keep the data returned by `_getexif` in the same (raw) format as pre 3.0.0? And return the new friendlier format in `getexif`. There's a lot of code out there that relies on the format of `_getexif`, going back to at least 2010.

**Comment by wiredfool** [comment]
So, the two tests that are failing at this point are frustrating. 

The whitepoint isn't roundtripping because it's coming back as an equivalent, but different fraction. The read metadata is tripping up with the difference between a float and the string repr of a float.

**Comment by wiredfool** [comment]
Ok, Tests are updated to slightly adjust the targeted values. As much as I hate doing that to get things to pass, I believe that it's valid for this set of tests.

**Pull request merged**